src/coreclr/vm/gc_unwind_x86.inl

```diff
@@ -0,0 +1,3890 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// This file is shared between CoreCLR and NativeAOT. Some of the differences are handled
+// with the FEATURE_NATIVEAOT and FEATURE_EH_FUNCLETS defines. There are three main methods
+// that are used by both runtimes - DecodeGCHdrInfo, UnwindStackFrameX86, and EnumGcRefsX86.
+//
+// The IN_EH_FUNCLETS, IN_EH_FUNCLETS_COMMA, and NOT_IN_NATIVEAOT_PRE_COMMA macros are used
+// to specify some parameters for the above methods that are specific for a certain runtime
+// or configuration.
+#ifdef FEATURE_EH_FUNCLETS
+#define IN_EH_FUNCLETS(a) a
+#define IN_EH_FUNCLETS_COMMA(a) a,
+#else
+#define IN_EH_FUNCLETS(a)
+#define IN_EH_FUNCLETS_COMMA(a)
+#endif
+#ifdef FEATURE_NATIVEAOT
+#define NOT_IN_NATIVEAOT_PRE_COMMA(a)
+#else
+#define NOT_IN_NATIVEAOT_PRE_COMMA(a) ,a
+#endif
```

Comment:
All runtime differences are abstracted with these macros and `#ifdef FEATURE_NATIVEAOT`.